tests/kernel/mem_protect/userspace/testcase.yaml

```diff
@@ -2,3 +2,4 @@ tests:
   kernel.memory_protection.userspace:
     filter: CONFIG_ARCH_HAS_USERSPACE
     tags: core security userspace ignore_faults
+    extra_sections: app_smem
```

Comment:
It's going to be cumbersome to have to update more and more YAML as tests are converted to use this.
Let's try to root-cause what's going on with sanitycheck,

Reply:
So, I inaccurately assumed it was sanity check.  Something in the way i have my directory structure currently causes the zephyr-env.sh to set the path for sanitycheck to a different instance of zephyr in a parent directory that did not have the modifications.  Thus requiring the modification to the yaml.